Source/ACE.Server/Factories/LootGenerationFactory.cs

```diff
@@ -1545,78 +1488,108 @@ private static int GetWorkmanship(int tier)
+                    break;
+                case ItemType.Gem:
                 default:
+                    minItemSpellCraftRange = 1.00f;
+                    maxItemSpellCraftRange = 1.00f;
                     break;
             }
 
-            return spellcraft;
+            // Getting the spell difficulty - Maybe a better way to do this.
+            var maxSpellLevel = wo.GetMaxSpellLevel();
+            int maxSpellDiff = 1;
+
+            if (maxSpellLevel == 2)
+                maxSpellDiff = 50;
+            else if (maxSpellLevel == 3)
+                maxSpellDiff = 100;
+            else if (maxSpellLevel == 4)
+                maxSpellDiff = 150;
+            else if (maxSpellLevel == 5)
+                maxSpellDiff = 200;
+            else if (maxSpellLevel == 6)
+                maxSpellDiff = 250;
+            else if (maxSpellLevel == 7)
+                maxSpellDiff = 300;
+            else if (maxSpellLevel == 8)
+                maxSpellDiff = 400;  
+
+            var tItemSpellCraft = maxSpellDiff * ThreadSafeRandom.Next(minItemSpellCraftRange, maxItemSpellCraftRange);
+
+            if (tItemSpellCraft < 0)
+                tItemSpellCraft = 0;
+
+            int finalItemSpellCraft = (int)Math.Floor(tItemSpellCraft);
+
+            return finalItemSpellCraft;
+
         }
 
-        private static int GetDifficulty(int tier, int spellcraft)
+        private static int GetDifficulty(WorldObject wo, int tier, int itemspellcraft)
         {
-            int difficulty = 0;
-            switch (tier)
-            {
-                case 1:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 2:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 3:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 4:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 5:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 6:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 7:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                case 8:
-                    difficulty = spellcraft + (ThreadSafeRandom.Next(0, 10) * ThreadSafeRandom.Next(1, 3));
-                    break;
-                default:
-                    break;
-            }
+            int wieldReq = 1;
+            int sc = 0;  
```

Comment:
some of these vars are confusing named, and don't provide much insight into what they are, without studying the code further

sc, spc, bq